src/engine/Server.cpp

```diff
@@ -870,10 +914,11 @@ net::awaitable<std::optional<Server::PlannedQuery>> Server::parseAndPlan(
   // probably related to issues in GCC's coroutine implementation.
   return computeInNewThread(
       [&query, &qec, enablePatternTrick = enablePatternTrick_,
-       handle = std::move(handle),
-       timeLimit]() mutable -> std::optional<PlannedQuery> {
+       handle = std::move(handle), timeLimit,
+       &additionalDatasets]() mutable -> std::optional<PlannedQuery> {
         auto pq = SparqlParser::parseQuery(query);
         handle->throwIfCancelled();
+        pq.datasetClauses_.addClauses(additionalDatasets);
```

Comment:
Are those te semantics dictated by the standard? Meaningthat you can combine the HTTP parameters and FROM [NAMED] clauses in the query?

Reply:
I'll also check myself when I have time

Reply:
The RDF dataset can be specified both in the query or the parameters. But what I didn't see before is that the dataset will be defined by only one of those at the time with a fixed precedence. I'll change that. This is specified in [SPARQL Protocol 2.1.4](https://www.w3.org/TR/2013/REC-sparql11-protocol-20130321/#dataset)